packages/jsii-pacmak/lib/targets/kotlin/kotlingenerator.ts

```diff
@@ -0,0 +1,254 @@
+import { GeneratorBase } from '../../generator';
+import path = require('path');
+import * as fs from 'fs-extra';
+import { createDeclaration, KTypeDeclaration, KElement } from './kotlinmodel';
+import { KotlinGeneratorConfiguration } from './kotlingeneratorconfiguration';
+import { DeclarationNode } from './declarationnode';
+import { Assembly, ReferenceType, EnumType, TypeSystem } from 'jsii-reflect';
+import { KotlinCodeMaker } from './kotlincodemaker';
+import { JvmAbi } from '../jvm/jvmabi';
+import { VERSION } from '../../version';
+import { KotlinTypeMapper } from './kotlintypemapper';
+
+export class KotlinGenerator extends GeneratorBase {
+  private static getFileName(fqn: string, system: TypeSystem): string {
+    const kotlinFqn = KotlinTypeMapper.mapReferenceName(fqn, system);
+    return `${path.join('src', 'main', 'kotlin', ...kotlinFqn.split('.'))}.kt`;
+  }
+
+  private static getAssemblyOutputDir(assembly: Assembly): string {
+    const dir = assembly.name.replace(/\./g, '/');
+    return path.join('src', 'main', 'resources', dir);
+  }
+
+  private readonly code = new KotlinCodeMaker();
+
+  private packageDir: string | undefined = undefined;
+  private _assembly: Assembly | undefined = undefined;
+
+  public constructor() {
+    super();
+  }
+
+  private get assembly(): Assembly {
+    const assembly = this._assembly;
+
+    if (!assembly) {
+      throw new Error('Assembly is not set');
+    }
+
+    return assembly;
+  }
+  
+  public async load(packageDir: string, assembly: Assembly): Promise<void> {
+    this.packageDir = packageDir;
+    this._assembly = assembly;
+    return Promise.resolve();
```

Comment:
You don't need that `return` clause at all here. I'll stop signaling the unnecessary `Promise`-wrapping of results of `async` functions from here one (I mean you probably got the point by now!)

Reply:
Hm-m, eslint says that "Async method 'load' has no 'await' expression.eslint(@typescript-eslint/require-await)" if I remove the line.
I think it makes sense to remove the `async` things altogether both from `KotlinGenerator` and `Generator`, as they don't benefit from the coroutine functionality.

Reply:
Yup you're correct, this pesky rule... Would be kinda nice if it could shut up about overrides/implementations... but well well no biggie.